web/src/components/apps/AppDetailPage.jsx

```diff
@@ -225,9 +228,11 @@ class AppDetailPage extends Component {
 
   componentDidMount() {
     const { history } = this.props;
+    console.log("appdetail page mounted");
 
     if (history.location.pathname === "/apps") {
-      this.checkForFirstApp();
```

Comment:
I'm not sure exactly why this didn't work in react 18. There's something about the order or number of times the lifecycle methods are called. For now we're just going to use this repeater to have this check happen outside of the lifecycle methods. eventually we will pull this completely out of the component and handle it outside of react.